src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,94 @@
+        saveToFile(report, toFileName);
+    }
+
+    private String createReport(Map<String, Integer> dataForReport) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(createLine(SUPPLY, dataForReport.get(SUPPLY)));
+        stringBuilder.append(createLine(BUY, dataForReport.get(BUY)));
+        stringBuilder.append(createLine(RESULT, dataForReport.get(RESULT)));
+        return stringBuilder.toString();
+    }
+
+    private String createLine(String key, Integer value) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(key)
+                .append(DATA_SEPARATOR)
+                .append(value)
+                .append(System.lineSeparator());
+        return stringBuilder.toString();
+    }
+
+    private void saveToFile(String report, String toFileName) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+            writer.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("can't write to the file", e);
+        }
+    }
+
+    private Map<String, Integer> createAggregatedData(List<String> lines) {
+        Map<String, Integer> dataforReport = new HashMap<>();
+        dataforReport.put(BUY, 0);
+        dataforReport.put(SUPPLY, 0);
+        for (String line : lines) {
+            String[] dataInLine = line.split(DATA_SEPARATOR);
+            if (dataInLine.length < 2) {
+                continue;
+            }
+            String key = dataInLine[0].trim();
+            if (!BUY.equals(key) && !SUPPLY.equals(key)) {
+                throw new RuntimeException(key + "is not a " + BUY + " or" + SUPPLY);
+            }
+            int value = 0;
+            try {
+                value = Integer.parseInt(dataInLine[1].trim());
+            } catch (NumberFormatException e) {
+                throw new RuntimeException(dataInLine[1] + "is not a number", e);
```

This violates checklist item "Don't ignore exceptions": the RuntimeException messages for invalid operation and number format (lines 61 and 67) are poorly formatted (missing spaces) and lack context. Include the offending line and input filename and preserve the original exception as the cause for parse errors. For example: `throw new RuntimeException("Invalid operation '" + key + "' in line '" + line + "' in file " + fromFileName);` and for number parsing include the NumberFormatException as the cause.